fitlins/base.py

```diff
@@ -83,9 +86,8 @@ def second_level(analysis, block, space, deriv_dir):
     analyses = []
 
     # pybids likes to give us a lot of extraneous columns
-    cnames = [contrast['name'] for contrast in block.contrasts]
     fmri_glm = level2.SecondLevelModel()
-    for contrasts, idx, ents in block.get_contrasts(names=cnames):
+    for contrasts, idx, ents in block.get_contrasts():
```

Comment:
Yes, previously the "variables" field was not doing anything. Now it will apply a "select" operation, so only the variables you care about are kept, and only contrasts for those variables will be produced here.